src/MLJGLMInterface.jl

```diff
@@ -327,4 +327,301 @@ metadata_model(
+to the case of a binary target variable, with a user-
+specified link function. Options exist to specify an intercept or offset feature.
+
+
+# Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+    mach = machine(model, X, y)
+
+Where
+
+- `X`: any table of input features (eg, a `DataFrame`) whose columns
+  each have one of the following element scitypes: `Continuous`,
+  `Count`, or `<:OrderedFactor`; check column scitypes with `schema(X)`
+
+- `y`: is the target, which can be any `AbstractVector` whose element
+  scitype is `<:OrderedFactor(2)` or `<:Multiclass(2)`; check the scitype
+  with `scitype(y)`
+
+# Hyper-parameters
+
+- `fit_intercept=true`: Whether to calculate the intercept for this model.
+   If set to false, no intercept will be calculated (e.g. the data is expected
+   to be centered)
+- `link = GLM.LogitLink`: The function which links the linear prediction function
+   to the probability of a particular outcome or class.
+- `offsetcol=nothing`: Name of the column to be used as an offset, if any.
+   An offset is a variable which is known to have a coefficient of 1.
+
+# Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given
+  features `Xnew` having the same scitype as `X` above. Predictions
+  are probabilistic.
+- `predict_mean(mach, Xnew)`: return predictions of the target given
+  features `Xnew`. Predictions represent probaility of being the
+  "positive" (second) class.
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `features`: The names of the features used during model fitting
+- `coef`: The linear coefficients determined by the model
+- `intercept`: The intercept determined by the model
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `deviance`: Measure of deviance of fitted model with respect to
+  a perfectly fitted model. For a linear model, this is the weighted
+  residual sum of squares
+- `dof_residual`: degrees of freedom for residuals, when meaningful
+- `stderror`: standard errors of the coefficients
+- `vcov`: estimated variance-covariance matrix of the coefficient estimates
+
+# Examples
+
+```
+using MLJ
+using CategoricalArrays
+using ScientificTypes
+CLF = @load LinearBinaryClassifier pkg=GLM
+clf = CLF(fit_intercept=false)
+
+X, y = @load_iris
+
+# needs binary classification
+idx = findall( x -> string(x) != "setosa", y)
+y = y[idx]
+scitype(y)
+droplevels!(y) # drop unused levels
+scitype(y) # need multiclass 2
+X = map( x -> x[idx], X)
+mach = machine(clf, X, y) |> fit!
+
+Xnew = (sepal_length = [6.4, 7.2, 7.4],
+        sepal_width = [2.8, 3.0, 2.8],
+        petal_length = [5.6, 5.8, 6.1],
+        petal_width = [2.1, 1.6, 1.9],)
+
+yhat = predict(mach, Xnew) # probabilistic predictions
+pdf(yhat, levels(y)) # probability matrix
+p_virginica = pdf.(yhat, "virginica")
+p_positive_class = predict_mean(mach, Xnew) # probability of "positive" class
```

Comment:
Not sure what you mean by "positive class" as this target is unordered (scitype `Multiclass{2}`, as opposed to `OrderedFactor{2}`). There is a "positive class" for `OrderedFactor{2}`.  See the discussion [here](https://alan-turing-institute.github.io/MLJ.jl/dev/working_with_categorical_data/#Binary-data). I think what you want to demonstrate here is `predict_mode`. I'm surprised `predict_mean` doesn't throw an error. Have you checked this code?

Users of other platforms are used to generating a probability matrix, but in MLJ you rarely actually need this. (And in the unordered case, there is an ambiguity about how to define this - one reason we predict distribution objects and not raw probability vectors). For example, all the metrics (aka measures) in MLJ expect "distribution objects" and never raw probabilities. This mitigates a lot of confusion about representation. So, no special casing for binary vs 3 or more classes. 

Maybe you output the display of `yhat[1]` and then `predict_mode(Xnew)[1]`. By the way `predict_mode(Xnew)` is syntactically the same as `mode.(predict(Xnew))`.